crates/bevy_ui/src/flex/mod.rs

```diff
@@ -269,7 +269,8 @@ pub fn flex_node_system(
         }
     }
 
-    if scale_factor_events.iter().next_back().is_some() || ui_scale.is_changed() {
+    if !scale_factor_events.is_empty() || ui_scale.is_changed() {
+        scale_factor_events.clear();
```

Comment:
`.next_back().is_some()` isn't the right way to do this, but unless multiple scale_factor_events are sent in a frame it behaves the same way.